src/relayserver/RelayServer.ts

```diff
@@ -594,50 +614,97 @@ export class RelayServer extends EventEmitter {
     }
-    const balance = toBN(await this.contractInteractor.getBalance(this.managerAddress))
+    const receipts: TransactionReceipt[] = []
+    const managerHubBalance = await this.relayHubContract?.balanceOf(this.managerAddress) ?? toBN(0)
     const gasPrice = await this.contractInteractor.getGasPrice()
+    const method = this.relayHubContract?.contract.methods.withdraw(toHex(managerHubBalance), this.owner)
+    const withdrawTxGasLimit = await method.estimateGas(
+      { from: this.getManagerAddress() })
+    const withdrawTxCost = toBN(withdrawTxGasLimit * parseInt(gasPrice))
+    // sending manager hub balance to owner
+    if (managerHubBalance.gte(withdrawTxCost)) {
+      console.log(`Sending manager hub balance ${managerHubBalance.toString()} to owner`)
+      receipts.push((await this._sendTransaction({
+        signerIndex: 0,
+        destination: this.relayHubContract?.address as string,
+        method
+      })).receipt)
+    } else {
+      console.log(`manager hub balance too low: ${managerHubBalance.toString()}, tx cost: ${withdrawTxCost.toString()}`)
+    }
+
     const gasLimit = mintxgascost
-    console.log(`Sending balance ${balance.div(toBN(1e18)).toString()} to owner`)
     const txCost = toBN(gasLimit * parseInt(gasPrice))
-    if (balance.lt(txCost)) {
-      throw new Error(`balance too low: ${balance.toString()}, tx cost: ${gasLimit * parseInt(gasPrice)}`)
-    }
-    const { receipt } = await this._sendTransaction({
-      signerIndex: 0,
-      destination: this.owner as string,
-      gasLimit: gasLimit.toString(),
-      gasPrice,
-      value: toHex(balance.sub(txCost))
-    })
+
+    const managerBalance = await this.getManagerBalance()
+    if (managerBalance.gte(txCost)) {
+      console.log(`Sending manager eth balance ${managerBalance.toString()} to owner`)
+      // sending manager eth balance to owner
+      receipts.push((await this._sendTransaction({
+        signerIndex: 0,
+        destination: this.owner as string,
+        gasLimit: gasLimit.toString(),
+        gasPrice,
+        value: toHex(managerBalance.sub(txCost))
+      })).receipt)
+    } else {
+      console.log(`manager balance too low: ${managerBalance.toString()}, tx cost: ${gasLimit * parseInt(gasPrice)}`)
+    }
+
+    // sending workers' balance to owner (currently one worker, todo: extend to multiple)
+    const workerBalance = await this.getWorkerBalance(1)
+    if (workerBalance.gte(txCost)) {
+      console.log(`Sending workers' eth balance ${workerBalance.toString()} to owner`)
+      receipts.push((await this._sendTransaction({
+        signerIndex: 1,
+        destination: this.owner as string,
+        gasLimit: gasLimit.toString(),
+        gasPrice,
+        value: toHex(workerBalance.sub(txCost))
+      })).receipt)
+    } else {
+      console.log(`balance too low: ${managerBalance.toString()}, tx cost: ${gasLimit * parseInt(gasPrice)}`)
```

Comment:
wrong error message: its worker, not manager.

Reply:
fixed